### PR TITLE
Display and measure room ambient noise with the microphone

### PR DESCRIPTION
**Pull request by jmettler27**
This feature adds the possibility to use its microphone to measure the ambient intensity (in decibels) of a room.
The decibels are displayed live when the user records the sound, and different colors help to indicate the level of disturbance.
This indication is then displayed for each room on their information fragment.

### CODE REVIEW

File: app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/RoomReviewInfoFragment.kt [hunk 1]
```diff
@@ -20,20 +29,82 @@ class RoomReviewInfoFragment : Fragment(R.layout.fragment_room_review_info) {
...
+        val intent = Intent(requireContext(), AudioRecordActivity::class.java)
+        getAudioResults.launch(intent)
     }
 
+    private val getAudioResults =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == Activity.RESULT_OK) {
+                val data: Intent? = result.data
+                val measure = data?.getDoubleExtra(AudioRecordActivity.EXTRA_MEASUREMENT_VALUE, 0.0)
+                if (measure != null && measure > 0.0) {
```
Comment (mbouilloud) [inline app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/RoomReviewInfoFragment.kt:101]: You check here that the measure is not 0 before uploading, but in the AudioRecordActivity you make a toast saying that the measure was successfully, even if the measure is 0.

File: app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/RoomReviewInfoFragment.kt [hunk 2]
```diff
@@ -20,20 +29,82 @@ class RoomReviewInfoFragment : Fragment(R.layout.fragment_room_review_info) {
...
+            )
+        }
+
+    }
+
+    private fun displayRoomNoise(noiseData: Map<String, Int>) {
+        if (noiseData.isEmpty()) return
+
+        val sortedMap = noiseData.toSortedMap()
+        val mostRecentDate = sortedMap.lastKey()
```
Comment (mbouilloud) [inline app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/RoomReviewInfoFragment.kt:73]: If we are just using the most recent recording, why keep all the older ones ?

File: app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/AudioRecordActivityTest.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    @get:Rule(order = 2)
+    val permissionRule: GrantPermissionRule = GrantPermissionRule.grant(
+        Manifest.permission.RECORD_AUDIO
+    )
+
+    @Before
+    fun setUp() {
+        val intent =
+            Intent(ApplicationProvider.getApplicationContext(), AudioRecordActivity::class.java)
```
Comment (mbouilloud) [inline app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/AudioRecordActivityTest.kt:43]: You don't need to launch an intent and use a scenario if you use a rule I think.

**Review by mbouilloud — APPROVED** [review]

File: app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/RoomReviewInfoFragment.kt (hunk 1 above)
Comment (jmettler27) [reply app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/RoomReviewInfoFragment.kt:101]: My bad, it should be possible to upload zero measures well. I will remove this strict condition.

File: app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/RoomReviewInfoFragment.kt (hunk 2 above)
Comment (jmettler27) [reply app/src/main/java/com/github/sdp/ratemyepfl/fragment/review/RoomReviewInfoFragment.kt:73]: I wanted to also display a small list of recent measures but I did not find time to implement it. I can do this in the next sprint.

**Comment by mbouilloud** [comment]
The CI problem may come from the fact that you changed Gradle version. If it doesn't work revert to the old version.

File: app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/AudioRecordActivityTest.kt (hunk 3 above)
Comment (jmettler27) [reply app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/AudioRecordActivityTest.kt:43]: Oh yes you're right! I will remove it.

**Review by AlexandreMessmer — APPROVED** [review]
Good job! I love this feature :)

**Pull request merged**